### ISSUE TITLE
Cannot install on php8.1 and symfony6

### ISSUE THREAD

**Issue by signmeuptwice** [PHP-FFMpeg/PHP-FFMpeg#842]
| Version     | not installed yet
php 8.1
symfony 6

### Support Question

cannot install `php-ffmpeg/php-ffmpeg` 

is it totally incompatible with my configuration ?

error message:

```
Your requirements could not be resolved to an installable set of packages.
 
   Problem 1
     - php-ffmpeg/php-ffmpeg[0.3.0, ..., 0.6.0] require alchemy/binary-driver ~1.5 -> satisfiable by alchemy/binary-driver[1.5.0, 1.6.0].
     - php-ffmpeg/php-ffmpeg[v0.17.0, ..., v0.18.0] require alchemy/binary-driver ^1.5 || ~2.0.0 || ^5.0 -> satisfiable by alchemy/binary-driver[1.5.0, 1.6.0, v2.0.0, 5.0.0, 5.1.0, v5.1.1, v5.2.0].
     - php-ffmpeg/php-ffmpeg[0.1.0, ..., 0.2.1] require symfony/process >=2.0,<=2.2 -> found symfony/process[2.0.4, ..., v2.2.0] but these were not loaded, likely because it conflicts with another require.
     - php-ffmpeg/php-ffmpeg[0.2.2, ..., 0.2.4] require symfony/process ~2.0 -> found symfony/process[2.0.4, ..., v2.8.52] but these were not loaded, likely because it conflicts with another require.
     - alchemy/binary-driver 1.5.0 require symfony/process ~2.0 -> found symfony/process[2.0.4, ..., v2.8.52] but these were not loaded, likely because it conflicts with another require.
     - php-ffmpeg/php-ffmpeg[0.6.1, ..., v0.16] require php ^5.3.9 || ^7.0 -> your php version (8.1.0) does not satisfy that requirement.
     - alchemy/binary-driver 1.6.0 requires symfony/process ^2.0|^3.0 -> found symfony/process[2.0.4, ..., v2.8.52, v3.0.0, ..., v3.4.47] but these were not loaded, likely because it conflicts with another require.
     - alchemy/binary-driver[v2.0.0, 5.0.0, ..., v5.1.1] require symfony/process ^2.3|^3.0|^4.0 -> found symfony/process[v2.3.0, ..., v2.8.52, v3.0.0, ..., v3.4.47, v4.0.0, ..., v4.4.35] but these were not loaded, likely because it conflicts with another require.
     - alchemy/binary-driver v5.2.0 requires symfony/process ^2.3|^3.0|^4.0|^5.0 -> found symfony/process[v2.3.0, ..., v2.8.52, v3.0.0, ..., v3.4.47, v4.0.0, ..., v4.4.35, v5.0.0, ..., v5.4.0] but these were not loaded, likely because it conflicts with another require.
     - Root composer.json requires php-ffmpeg/php-ffmpeg * -> satisfiable by php-ffmpeg/php-ffmpeg[0.1.0, ..., v0.18.0].
```

my composer.json

```
{
    "type": "project",
    "license": "proprietary",
    "minimum-stability": "stable",
    "prefer-stable": true,
    "require": {
        "php": ">=8.1.0",
        "ext-ctype": "*",
        "ext-fileinfo": "*",
        "ext-gd": "*",
        "ext-iconv": "*",
        "ext-imagick": "^3.6",
        "ext-json": "*",
        "ext-openssl": "*",
        "doctrine/doctrine-bundle": "^2.5",
        "doctrine/orm": "^2.10",
        "jms/serializer-bundle": "^4.0",
        "knplabs/knp-paginator-bundle": "^5.8",
        "symfony/asset": "6.0.*",
        "symfony/cache": "6.0.*",
        "symfony/config": "6.0.*",
        "symfony/console": "6.0.*",
        "symfony/dependency-injection": "6.0.*",
        "symfony/dotenv": "6.0.*",
        "symfony/flex": "^2",
        "symfony/framework-bundle": "6.0.*",
        "symfony/mailer": "6.0.*",
        "symfony/mime": "6.0.*",
        "symfony/runtime": "6.0.*",
        "symfony/twig-bundle": "6.0.*",
        "symfony/validator": "6.0.*",
        "symfony/web-profiler-bundle": "6.0.*",
        "symfony/yaml": "6.0.*"
        "php-ffmpeg/php-ffmpeg": "*"
    },
    "require-dev": {
        "symfony/maker-bundle": "^1.36",
        "symfony/var-dumper": "6.0.*"
    },
    "config": {
        "optimize-autoloader": true,
        "preferred-install": {
            "*": "dist"
        },
        "sort-packages": true
    },
    "autoload": {
        "psr-4": {
            "App\\": "src/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "App\\Tests\\": "tests/"
        }
    },
    "replace": {
        "symfony/polyfill-ctype": "*",
        "symfony/polyfill-iconv": "*",
        "symfony/polyfill-php72": "*",
        "symfony/polyfill-php73": "*",
        "symfony/polyfill-php74": "*",
        "symfony/polyfill-php80": "*"
    },
    "scripts": {
        "auto-scripts": {
            "cache:clear": "symfony-cmd",
            "assets:install %PUBLIC_DIR%": "symfony-cmd"
        },
        "post-install-cmd": [
            "@auto-scripts"
        ],
        "post-update-cmd": [
            "@auto-scripts"
        ]
    },
    "conflict": {
        "symfony/symfony": "*"
    },
    "extra": {
        "symfony": {
            "allow-contrib": false,
            "require": "6.0.*"
        }
    }
}

```

**Comment by pascalbaljet** [comment]
Thanks for reporting @signmeuptwice. I've opened a PR for the underlying library: https://github.com/alchemy-fr/BinaryDriver/pull/47

**Comment by azurre** [comment]
@pascalbaljet It looks like alchemy-fr/BinaryDriver is dead. Last change was 2 years ago.

**Comment by secit-pl** [comment]
Could the alchemy-fr/BinaryDriver be replaced with symfony/process?

**Comment by pascalbaljet** [comment]
A few weeks before I became a maintainer of this repo, I started working on a modernized fork:
https://github.com/protonemedia/php-ffmpeg/

The packages `alchemy/binary-driver`, `evenement/evenement`, and `neutron/temporary-filesystem` are removed or adapted, and there's only support for PHP 8.0 and higher.

There are probably some breaking changes, but I'd love to merge the modernized fork into this repo.

**Comment by UtechtDustin** [comment]
@pascalbaljet thank you for the good work, but i still can't install your version with Symfony 6 and PHP 8.1

```
Using version ^1.1 for protonemedia/php-ffmpeg
./composer.json has been updated
Running composer update protonemedia/php-ffmpeg
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires protonemedia/php-ffmpeg ^1.1 -> satisfiable by protonemedia/php-ffmpeg[v1.1.0].
    - protonemedia/php-ffmpeg v1.1.0 requires doctrine/cache ^1.0 -> found doctrine/cache[v1.0, ..., 1.12.1] but the package is fixed to 2.1.1 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
You can also try re-running composer require with an explicit version constraint, e.g. "composer require protonemedia/PHP-FFMpeg:*" to figure out if any version is installable, or "composer require protonemedia/PHP-FFMpeg:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
```

**Comment by pascalbaljet** [comment]
I've opened a PR to merge my fork into this repo: https://github.com/PHP-FFMpeg/PHP-FFMpeg/pull/849

**Comment by UtechtDustin** [comment]
@pascalbaljet i got the error message from you fork :D

**Comment by pascalbaljet** [comment]
Yep fixed the errors :)

**Comment by UtechtDustin** [comment]
Tested it, seems working.
Thank you!

**Comment by pascalbaljet** [comment]
Fixed in [v1.0.0](https://github.com/PHP-FFMpeg/PHP-FFMpeg/releases/tag/v1.0.0).
